### PR TITLE
AA multisig tutorial: language refinement, compiler/libs update, ensure working

### PR DESCRIPTION
**Pull request by kailash-manasarovar**
- Full copy edit of language.
- Full check of instruction flow.
- Updated compiler version to 1.3.8.
- Managing working compiler/library versions so tutorial is functional.
- Addition of [DOC-96](https://linear.app/matterlabs/issue/DOC-96/emphasize-the-importance-of-the-factory-in-aa).
- Addition of [DOC-237](https://linear.app/matterlabs/issue/DOC-237/checking-now-update-aa-multisig-tutorial-with-comments-from-stas)

### CODE REVIEW

File: docs/dev/tutorials/custom-aa-tutorial.md [hunk 1]
```diff
@@ -334,35 +329,42 @@ function isValidSignature(bytes32 _hash, bytes memory _signature)
...
 
 ### Transaction validation
 
-Let's implement the validation process. It is responsible for validating the signature of the transaction and incrementing the nonce. Note, that there are some limitations on what this method is allowed to do. You can read more about them [here](../developer-guides/aa.md#limitations-of-the-verification-step).
+The transaction validation process is responsible for validating the signature of the transaction and incrementing the nonce. 
 
-To increment the nonce, you should use the `incrementNonceIfEquals` method of the `NONCE_HOLDER_SYSTEM_CONTRACT` system contract. It takes the nonce of the transaction and checks whether the nonce is the same as the provided one. If not, the transaction reverts. Otherwise, the nonce is increased.
+:::info
+- There are [some limitations](../developer-guides/aa.md#limitations-of-the-verification-step) on this function.
+:::
```
Comment (krofax) [inline docs/dev/tutorials/custom-aa-tutorial.md:340]:
```suggestion
:::info
- There are some [limitations](../developer-guides/aa.md#limitations-of-the-verification-step) on this function.
:::
```

File: docs/dev/tutorials/custom-aa-tutorial.md [hunk 2]
```diff
@@ -334,35 +329,42 @@ function isValidSignature(bytes32 _hash, bytes memory _signature)
...
+Even though the requirements above mean the accounts only touch their own storage slots, accessing your nonce in the `NONCE_HOLDER_SYSTEM_CONTRACT` is a [whitelisted](../developer-guides/aa.md#extending-the-set-of-slots-that-belong-to-a-user) case, since it behaves in the same way as your storage, it just happens to be in another contract. 
 
-Note that since the non-view methods of the `NONCE_HOLDER_SYSTEM_CONTRACT` are required to be called with the `isSystem` flag on, the [systemCallWithPropagatedRevert](https://github.com/matter-labs/v2-testnet-contracts/blob/main/l2/system-contracts/libraries/SystemContractsCaller.sol#L75) method of the `SystemContractsCaller` library should be used, so this library needs to be imported as well:
+To call the `NONCE_HOLDER_SYSTEM_CONTRACT`, we add the following import:
 
 ```solidity
-// to call non-view method of system contracts
-import "@matterlabs/zksync-contracts/l2/system-contracts/libraries/SystemContractsCaller.sol";
+// Access zkSync system contracts, in this case for nonce validation vs NONCE_HOLDER_SYSTEM_CONTRACT
+import "@matterlabs/zksync-contracts/l2/system-contracts/Constants.sol";
```
Comment (krofax) [inline docs/dev/tutorials/custom-aa-tutorial.md:350]: @kailash-manasarovar this statement/comment isn't clear.
Is there suppose to be a link here?

File: docs/dev/tutorials/custom-aa-tutorial.md [hunk 3]
```diff
@@ -334,35 +329,42 @@ function isValidSignature(bytes32 _hash, bytes memory _signature)
...
+- Use the [systemCallWithPropagatedRevert](https://github.com/matter-labs/v2-testnet-contracts/blob/main/l2/system-contracts/libraries/SystemContractsCaller.sol#L75) function of the `SystemContractsCaller` library.
+- Import this library also:
+    ```solidity
+    // to call non-view function of system contracts
+    import "@matterlabs/zksync-contracts/l2/system-contracts/libraries/SystemContractsCaller.sol";
+    ```
+:::
 
-The `TransactionHelper` library (already imported above with `using TransactionHelper for Transaction;`) can be used to get the hash of the transaction that should be signed. You can also implement your own signature scheme and use a different commitment for the transaction to sign, but in this example we use the hash provided by this library.
+Use the `TransactionHelper` library (already imported above with `using TransactionHelper for Transaction;`) to get the hash of the transaction that should be signed. You can also implement your own signature scheme and use a different commitment for signing the transaction, but in this example we use the hash provided by this library.
```
Comment (krofax) [inline docs/dev/tutorials/custom-aa-tutorial.md:363]:
```suggestion
Use the `TransactionHelper` library, as imported above with `using TransactionHelper for Transaction;` to get the transaction hash that should be signed. You can also implement your own signature scheme and use a different commitment for signing the transaction, but in this example, we use the hash provided by this library.
```

File: docs/dev/tutorials/custom-aa-tutorial.md [hunk 4]
```diff
@@ -781,25 +795,22 @@ contract AAFactory {
         (accountAddress) = abi.decode(returnData, (address));
     }
 }
-
 ```
 
-It's worth remembering that on zkSync, contract deployments are not done via bytecode, but via bytecode hash. The bytecode itself is passed to the operator via `factoryDeps` field. Note, that the `_aaBytecodeHash` must be formed specially:
+It's worth remembering that, on zkSync, contract deployments are not done via bytecode, but via bytecode hash. The bytecode itself is passed to the operator via the `factoryDeps` field. Note that the `_aaBytecodeHash` must be formed in the following manner:
```
Comment (krofax) [inline docs/dev/tutorials/custom-aa-tutorial.md:800]:
```suggestion
It's worth remembering that, on zkSync Era, [contract deployments](../building-on-zksync/contracts/contract-deployment.md)  are not done via bytecode, but via bytecode hash. The bytecode itself is passed to the operator via the `factoryDeps` field. Note that the `_aaBytecodeHash` must be formed in the following manner:
```

File: docs/dev/tutorials/custom-aa-tutorial.md [hunk 5]
```diff
@@ -781,25 +795,22 @@ contract AAFactory {
...
+This functionality is built into the SDK.
 
-## Deploying the factory
+## Deploy the factory
 
 ::: tip
-
-Make sure you deposit funds on zkSync using the [zkSync Portal](https://goerli.portal.zksync.io/bridge) before running your deployment scripts.
-
+Make sure you deposit funds on zkSync using the [zkSync Portal](https://goerli.portal.zksync.io/bridge) before running the deployment script.
```
Comment (krofax) [inline docs/dev/tutorials/custom-aa-tutorial.md:810]:
```suggestion
Make sure you deposit funds on zkSync Era using the [zkSync Portal](https://goerli.portal.zksync.io/bridge) before running the deployment script.
```

File: docs/dev/tutorials/custom-aa-tutorial.md [hunk 6]
```diff
@@ -850,10 +861,15 @@ Note that the address will be different for each run.
 
 ### Deploying an account
 
-Now, let's deploy an account and initiate a new transaction with it. In this section, we assume that you already have an EOA account with enough funds on zkSync.
-In the `deploy`, folder creates a file `deploy-multisig.ts`, where we will put the script.
+Now, let's deploy an account and use it to initiate a new transaction. 
+
+:::warning
+This section assumes you have an EOA account with sufficient funds on zkSync.
+:::
```
Comment (krofax) [inline docs/dev/tutorials/custom-aa-tutorial.md:868]:
```suggestion
:::warning
This section assumes you have an EOA account with sufficient funds on zkSync Era.
:::
```

File: docs/dev/tutorials/custom-aa-tutorial.md (hunk 2 above)
Comment (kailash-manasarovar) [reply docs/dev/tutorials/custom-aa-tutorial.md:350]: No link, **vs** should be **via**, can you suggest it please @krofax ?

Comment (kailash-manasarovar) [reply docs/dev/tutorials/custom-aa-tutorial.md:350]: If you look two paragraphs up on the first entry, they explain it. I've changed `vs` to `via`.

**Review by krofax — APPROVED** [review]

**Pull request merged**